### PR TITLE
[Refactor] Refactor RunMode for running UT easiser

### PR DESCRIPTION
**Pull request by sduzh**
- Move RunMode out of GlobalStateMgr, make it's easier to mock the run mode in unit test.
- Support hybrid mode, for test purpose.

## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto backported to target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

**Review by abc982627271 — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 37 / 50 (74.00%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/server/NodeMgr.java|0|1|00.00%|[364]|
|:large_blue_circle:|com/starrocks/catalog/ColocateTableIndex.java|0|1|00.00%|[1013]|
|:large_blue_circle:|com/starrocks/StarRocksFE.java|0|1|00.00%|[135]|
|:large_blue_circle:|com/starrocks/sql/analyzer/CreateTableAnalyzer.java|2|4|50.00%|[80, 84]|
|:large_blue_circle:|com/starrocks/server/RunMode.java|20|28|71.43%|[44, 45, 46, 47, 48, 50, 51, 77]|
|:large_blue_circle:|com/starrocks/common/proc/BackendsProcDir.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/sql/common/EngineType.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/system/SystemInfoService.java|2|2|100.00%|[]|
|:large_blue_circle:|com/starrocks/system/ComputeNode.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/server/GlobalStateMgr.java|6|6|100.00%|[]|
|:large_blue_circle:|com/starrocks/common/util/PropertyAnalyzer.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/transaction/PublishVersionDaemon.java|1|1|100.00%|[]|
|:large_blue_circle:|com/starrocks/system/HeartbeatMgr.java|1|1|100.00%|[]|

**Review by kevincai — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by Seaven — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

File: fe/fe-core/src/main/java/com/starrocks/common/util/PropertyAnalyzer.java [hunk 1]
```diff
@@ -385,7 +386,7 @@ private static void checkReplicationNum(short replicationNum) throws AnalysisExc
         // compute-storage-separation table will be created and in this case the replication_num will
         // be ignored, so there is no need to check whether the number of alive nodes is greater than the
         // replication_num.
-        if (GlobalStateMgr.getCurrentState().isSharedNothingMode()) {
+        if (!RunMode.getCurrentRunMode().isIgnoreReplicationNum()) {
```
Comment (wyb) [inline fe/fe-core/src/main/java/com/starrocks/common/util/PropertyAnalyzer.java:389]: create lake table in HYBRID run mode does not need to check replication number.

Comment (sduzh) [reply fe/fe-core/src/main/java/com/starrocks/common/util/PropertyAnalyzer.java:389]: Yes I know, need some refactoring here. Will do that in a separate patch.

**Comment by sduzh** [comment]
@Mergifyio backport branch-3.0

**Pull request merged**